### PR TITLE
Module improvements round 2

### PR DESCRIPTION
**Pull request by JerrySmidt**
Mostly a rewrite of the Javascript into Magento UI components.

* International autocomplete is now a separate field.
* Add descriptive label to international autocomplete.
* Add a label to house number select.
* Fix blank option in house number select.
* Fix different styling of error messages.
* Use latest version (1.2) of Postcode autocomplete library.
* Use AMD to import autocomplete library.
* Fix API autocomplete response format.
* Add loading indication to getDetails call.
* Fix field sortorder.
* Add dependency on checkout module.
* Update module version to 3.0.0.

### CODE REVIEW

**Comment by jorgb90** [comment]
@JozefVerhoef Would be great if you could verify and accept these changes.

File: composer.json [hunk 1]
```diff
@@ -2,7 +2,9 @@
     "name": "postcode-nl/api-magento2-module",
     "description": "Postcode.nl International Address API module for Magento 2. Adds autocompletion for addresses in multiple countries using official postal data.",
     "require": {
-      "php": "7.*|~7.0.6"
+      "php": "7.*|~7.0.6",
+      "magento/module-checkout": "100.2.*",
+      "magento/module-ui": "101.0.*"
```
Comment (barryvdh) [inline composer.json:7]: Magento 2.4.2 is 101.2.2, so this means this module cannot be installed with Magento 2.4 right now.

**Comment by barryvdh** [comment]
The version constraints in this PR make it impossible to install on the latest Magento (2.4.2).

File: composer.json (hunk 1 above)
Comment (JerrySmidt) [reply composer.json:7]: Thanks Barry, the dependency will be fixed in the next release.

**Pull request merged**